node/bft/events/src/block_response.rs

```diff
@@ -40,16 +46,18 @@ impl<N: Network> EventTrait for BlockResponse<N> {
 impl<N: Network> ToBytes for BlockResponse<N> {
     fn write_le<W: Write>(&self, mut writer: W) -> IoResult<()> {
         self.request.write_le(&mut writer)?;
-        self.blocks.write_le(&mut writer)
+        self.blocks.write_le(&mut writer)?;
+        self.latest_consensus_version.write_le(&mut writer)
```

Comment:
As the message is not versioned, upon upgrade, other nodes immediately won't be able to parse this message and sync from the upgraded node anymore.

As validatord and their core clients always upgrade first, this effectively means the rest of the network will halt until they upgrade.

Can/should we make this field Optional, and only write it once we are at a new ConsensusVersion? Note that if we do this, we also need a backwards compatible way to determine the message version when reading. Similar to the recent PR by Lukasz adjusting PeerResponse serialization, perhaps this can be handled by assuming an empty request (first bytes of this payload) is an implicit version indicator?

Reply:
For reference, this is how it was done with the `PeerResponse` extension: https://github.com/ProvableHQ/snarkOS/pull/3860.

Reply:
V12 is merged now